File: src/main/java/myLang/block/statement/IfThenStatementContext.java
package myLang.block.statement;

import myLang.ContextHandler;
import myLang.expression.ExpressionContext;
import myLang.response.BooleanResponse;
import myLang.response.EmptyResponse;
import myLang.response.MyLangException;
import myLang.response.Response;
import myLangParser.MyLangParser;

public class IfThenStatementContext implements ContextHandler<MyLangParser.IfThenStatementContext> {
    @Override
    public Response handler(MyLangParser.IfThenStatementContext ctx) throws MyLangException {
        Response response = new ExpressionContext().handler(ctx.expression());
        if(!(response instanceof BooleanResponse)){
            throw new MyLangException("Оператор if требует тип boolean");
        }
        if((boolean)response.getResponse()){
            return new StatementContext().handler(ctx.statement());
        }
        return new EmptyResponse();
    }
}

File: src/main/kotlin/myLangKotlin/block/statement/IfThenStatementContext.kt
package myLangKotlin.block.statement

import myLangKotlin.ContextHandler
import myLangKotlin.expression.ExpressionContext
import myLangKotlin.response.BooleanResponse
import myLangKotlin.response.EmptyResponse
import myLangKotlin.response.MyLangException
import myLangKotlin.response.Response
import myLangParser.MyLangParser

class IfThenStatementContext : ContextHandler<MyLangParser.IfThenStatementContext> {
    @Throws(MyLangException::class)
    override fun handler(ctx: MyLangParser.IfThenStatementContext): Response<*> {
        val response = ExpressionContext().handler(ctx.expression()) as? BooleanResponse
                ?: throw MyLangException("Оператор if требует тип boolean")
        return if (response.response as Boolean) {
            StatementContext().handler(ctx.statement())
        } else EmptyResponse()
    }
}
